File: src/main/java/com/codingspezis/android/metalonly/player/plan/PlanItem.java
package com.codingspezis.android.metalonly.player.plan;

/**
 * Represents a plan entry.
 * <p/>
 * Base class for {@link PlanRealEntryItem}, {@link PlanSectionItem}
 */
public abstract class PlanItem {

    protected PlanData data = null;

    public PlanData getPlanData() {
        return data;
    }

    public abstract boolean isSection();

}

File: src/main/kotlin/com/codingspezis/android/metalonly/player/plan/PlanItem.kt
package com.codingspezis.android.metalonly.player.plan

/**
 * Represents a plan entry.
 *
 *
 * Base class for [PlanRealEntryItem], [PlanSectionItem]
 */
abstract class PlanItem {

    @JvmField
    internal var planData: ShowInformation? = null

    abstract val isSection: Boolean

}